src/main/java/picard/sam/SamToFastqWithTags.java

```diff
@@ -0,0 +1,213 @@
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.fastq.FastqRecord;
+import htsjdk.samtools.fastq.FastqWriter;
+import htsjdk.samtools.fastq.FastqWriterFactory;
+import htsjdk.samtools.util.IOUtil;
+import htsjdk.samtools.util.Log;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import picard.PicardException;
+import picard.cmdline.programgroups.ReadDataManipulationProgramGroup;
+
+import java.io.File;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@CommandLineProgramProperties(
+        summary = SamToFastqWithTags.USAGE_SUMMARY + SamToFastqWithTags.USAGE_DETAILS,
+        oneLineSummary = SamToFastqWithTags.USAGE_SUMMARY,
+        programGroup = ReadDataManipulationProgramGroup.class)
+public class SamToFastqWithTags extends SamToFastq {
+    static final String USAGE_SUMMARY = "Converts a SAM or BAM file to FASTQ.";
+    static final String USAGE_DETAILS = "This will create the fastq from read sequences alongside two fastq files whose sequence is generated from tags.  " +
+            "One will be converted from the \"CR\" tag with quality from the \"CY\" tag.  The other fastq will be " +
+            "converted from the \"CB\" and \"UR\" tags concatenated together with no separator (not specified on " +
+            "command line) with the qualities coming from the \"CY\" and \"UY\" tags concatenated together.  The two files " +
+            "will be named CR.fastq and CB_UR.fastq" +
+            "<br />" +
+            "<pre>" +
+            "java -jar picard.jar SamToFastqWithTags <br />" +
+            "     I=input.bam<br />" +
+            "     FASTQ=output.fastq<br />" +
+            "     SEQUENCE_TAG_GROUP=CR<br />" +
+            "     QUALITY_TAG_GROUP=CY<br />" +
+            "     SEQUENCE_TAG_GROUP=\"CB,UR\"<br />" +
+            "     QUALITY_TAG_GROUP=\"CY,UY\"" +
+            "</pre>" +
+            "<hr />";
+
+    @Argument(shortName = "STG", doc = "List of comma separated tag values to extract from Input SAM/BAM to be used as read sequence", minElements = 1)
+    public List<String> SEQUENCE_TAG_GROUP;
+
+    @Argument(shortName = "QTG", doc = "List of comma separated tag values to extract from Input SAM/BAM to be used as read qualities", optional = true)
+    public List<String> QUALITY_TAG_GROUP;
+
+    @Argument(shortName = "SEP", doc = "List of sequences to put in between each comma separated list of sequence tags in each SEQUENCE_TAG_GROUP (STG)", optional = true)
+    public List<String> TAG_GROUP_SEPERATOR;
+
+    @Argument(shortName = "GZOPTG", doc = "Compress output FASTQ files per Tag grouping using gzip and append a .gz extension to the file names.")
+    public Boolean COMPRESS_OUTPUTS_PER_TAG_GROUP = false;
+
+    private final Log log = Log.getInstance(SamToFastqWithTags.class);
+
+    private final static String TAG_SPLIT_DEFAULT_SEP = "";
```

Comment:
Correct modifier ordering is `private static final`; the IDE should highlight this and sort them for you.

Reply:
my ide does not but changed